Sources/NIO/EventLoop.swift

```diff
@@ -291,6 +295,106 @@ extension TimeAmount: Comparable {
     }
 }
 
+extension TimeAmount {
+    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds + rhs.nanoseconds)
+    }
+
+    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds - rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount.Value, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs * rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount, rhs: TimeAmount.Value) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds * rhs)
+    }
+
+    public static func * (lhs: Int, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(TimeAmount.Value(lhs) * rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount, rhs: Int) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds * TimeAmount.Value(rhs))
+    }
+}
+
+/// Represents a point in time.
+///
+/// Stores the time in nanoseconds as returned by `DispatchTime.now().uptimeNanoseconds`
+///
+/// `NIODeadline` allow chaining multiple tasks with the same deadline without needing to
+/// compute new timeouts for each step
+///
+/// ```
+/// func doSomething(deadline: NIODeadline) -> EventLoopFuture<Void> {
+///     return step1(deadline: deadline).then {
+///         step2(deadline: deadline)
+///     }
+/// }
+/// doSomething(deadline: .now() + .seconds(5))
+/// ```
+///
+/// - note: `NIODeadline` should not be used to represent a time interval
+public struct NIODeadline: Equatable, Hashable {
+    public typealias Value = UInt64
+
+    /// The nanoseconds representation of the `NIODeadline`.
+    public let nanoseconds: Value
```

Comment:
this should be called something else I think. Maybe `uptimeNanoseconds` too? Otherwise people think that they can save this number and re-create a new `NIODeadline` from it elsewhere. But that's only true on the same computer and the same boot.

Reply:
Agreed.